src/utils.ts

```diff
@@ -5,6 +5,20 @@ import { v4 as uuidv4 } from 'uuid';
 import defaults from './defaults.json' with { type: 'json' };
 import { OrganicResult, TimeMeasure, UserData } from './types.js';
 
+const PROTOCOL_REGEX = /^(https?|ftp):\/\//;
+const DOMAIN_REGEX = /^([a-zA-Z0-9-]+\.)+[a-zA-Z]{2,6}(\/[a-zA-Z0-9._~:/?#[\]@!$&'()*+,;=%-]*)?$/;
+
+const URL_REGEX = new RegExp(
```

Comment:
Do we really need this? Why not use `URL.parse()`? BTW we should only consider it a URL if it starts with `http://` or `https://` (and not others like `ftp://`), e.g. consider "test.io" a search phrase, not URL